File: app/src/main/java/com/example/imagemachine/feature/launch/presenter/LaunchPresenter.java
package com.example.imagemachine.feature.launch.presenter;

import com.example.imagemachine.feature.base.presenter.BasePresenter;
import com.example.imagemachine.feature.launch.view.ILaunchView;

public class LaunchPresenter<V extends ILaunchView> extends BasePresenter<V> implements ILaunchPresenter {

    public LaunchPresenter(V view) {
        super(view);
    }

    @Override
    public void viewDidAttach() {
        super.viewDidAttach();
    }

    @Override
    public void viewDidDetach() {
        super.viewDidDetach();
    }
}

File: app/src/main/java/com/example/imagemachine/feature/launch/presenter/LaunchPresenter.kt
package com.example.imagemachine.feature.launch.presenter

import com.example.imagemachine.feature.base.presenter.BasePresenter
import com.example.imagemachine.feature.launch.view.ILaunchView

internal class LaunchPresenter<V : ILaunchView>
internal constructor(view: V) : BasePresenter<V>(view), ILaunchPresenter<V> {

    override fun viewDidAttach() {
        super.viewDidAttach()
    }

    override fun viewDidDetach() {
        super.viewDidDetach()
    }
}